packages/components/src/Form/Inputs/InputFilters/InputFilters.tsx

```diff
@@ -49,41 +50,58 @@ export interface FieldFilter {
 
 export interface InputFiltersProps {
   filters: FieldFilter[]
+  onChange: (filters: FieldFilter[]) => void
   hideFilterIcon?: boolean
   className?: string
 }
 
 const InputFiltersLayout: FC<InputFiltersProps> = ({
   className,
   filters,
+  onChange,
   hideFilterIcon = false,
 }) => {
-  const assigned = filters.filter((filter) => filter.value)
-
   const [fieldEditing, setFieldEditing] = useState<undefined | string>(
     undefined
   )
-  const [assignedFilters, setAssignedFilters] = useState(assigned)
   const [filterLookupName, setFilterLookupName] = useState('')
 
+  const assignedFilters = filters.filter(
+    (filter) => filter.value || filter.field === fieldEditing
+  )
```

Comment:
My bad, my previous suggestion resulted in a funky behavior where the filter tokens would end up in their "original" order, instead of the order the user selects them. This change should fix that:
```suggestion
  const assignedFilters = filters
    .filter((filter) => filter.value || filter.field === fieldEditing)
    .sort((a, b) =>
      a.value === undefined ? 1 : b.value === undefined ? -1 : 0
    )
```